Source/UI/NewUI/BillingForm/ViewModel/DefaultBillingFormViewModel.swift

```diff
@@ -168,14 +168,15 @@ final class DefaultBillingFormViewModel: BillingFormViewModel {
     
     private func validateTextOnEndEditing(textField: BillingFormTextField) {
         guard let type = textField.type else { return }
+        guard let textField = textField as? UITextField else { return }
```

Comment:
I am not sure this line reads well at all. In my mind it reads as

> Ensure textfield is an UITextField from textfield

I wouldn't worry too much about the syntactic aspect, but more of how we got here. Why a property textfield, of type BillingFormTextField, needs to be checked that it is in fact a UITextField ? As engineers in the Apple ecosystem, I think we have an expectation that once something is textField, it inherits (at least at some point) from `UITextField`.

Can't give an accurate suggestion on what to change here, but I would like you to review if given this, do you think there is another issue being created somewhere else, leading to this _symptom_ .

Reply:
Agree with Alex here I may be missing something But quickly looking at the code base looks like BillingFormTextField is class inheriting from UITextField already. 

`class BillingFormTextField: UITextField {`

So if you were to delete line 171 will that still not work ? as it might not require the typecast to UITextField